frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { Popper } from '@console/shared';
+import './Spotlight.scss';
+
+type SpotlightProps = {
+  selector: string;
+  interactive?: boolean;
+};
+
+const isInViewport = (element) => {
+  const rect = element.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+};
+
+const Spotlight: React.FC<SpotlightProps> = ({ selector, interactive }) => {
+  const element = document.querySelector(selector);
+  const { height, width } = element.getBoundingClientRect();
+  const style: React.CSSProperties = {
+    height,
+    width,
+  };
+
+  const [clicked, setClicked] = React.useState(false);
+  React.useEffect(() => {
+    if (!isInViewport(element)) {
+      element.scrollIntoView();
+    }
+    if (!interactive) {
+      const handleClick = () => setClicked(true);
+      document.addEventListener('mousedown', handleClick);
+      return () => {
+        document.removeEventListener('mousedown', handleClick);
+      };
+    }
+    return () => {};
+  }, [clicked, element, interactive]);
+
+  const highlight = (
+    <div
+      className={cx('ocs-spotlight', {
+        'ocs-spotlight__element-highlight-animate': interactive && !clicked,
+        'ocs-spotlight__element-highlight-noanimate': !interactive,
+      })}
+      style={style}
+    />
+  );
+  return (
+    <Popper reference={element} placement="top-start">
```

Comment:
Your popper is trying to be too smart... it's pushing away from the edge of the screen.

```suggestion
    <Popper reference={element} placement="top-start" popperOptions={popperOptions}>
```

```
  import { PopperOptions } from 'popper.js';
  ...
  const popperOptions: PopperOptions = {
    modifiers: {
      preventOverflow: {
        boundariesElement: 'window',
        padding: 0,
      },
    },
  };
```

Reply:
updated.